docs/dusty-specs/app-specs.md

```diff
@@ -0,0 +1,132 @@
+# Dusty App Specs
+These specs are used to specify apps for you local environment. They should go in the apps folder in your specs repo.
+
+## Keys
+
+### repo
+```
+repo: github.com/gamechanger/app1
+repo: /Users/dusty/app1
+```
+This repo key is used to specify a local or remote github repo where the code for the app lives.
+
+### mount
+```
+mount: /gc/app1
+```
+The mount key is used to specify where in the container's file system to mount the code specified by the repo key.  The mounting is done using docker's volume mounts. <br />
+Using the image and mount keys specified above, the code specified in `/Users/dusty/app1` on your mac will be visible in your container at `/gc/app1`.<br />
+The repo and mount keys are linked.  If you specify one you must specify the other. If you do not specify one, you cannot specify the other.<br />
+
+
+### depends
+```
+depends:
+  services:
+    - coreMongo
+    - coreRedis
+  apps:
+    - deferrableworker
+    - docketsworker
+  libs:
+    - gclib
+```
+The depends key is used to specify what libs, apps and services this app depends on.<br />
+Apps and services specified in the depends dict are spun up before the current app. The current app is then linked (using docker-compose's links) to the apps and services. Because we are using docker-compose links, dependencies must only be 1 way.  This means you cannot have app1 depend on app2 and app2 depend on app1.<br />
+Libs specified in the depends dict have a different meaning.  They tell dusty that these libs (and their recursive dependencies) need to be mounted into this app's container (using docker's volume mounts).  In so doing, the running app will have access to the most recent code in these libs.<br />
+The values corresponding to the services, apps, and libs keys are all lists.<br />
+
+### conditional_links
+```
+conditional_links:
+  apps:
+    - app2
+```
+The conditional_links key is similar to the depends key. There are some big differences though. First, the conditional_links key can only specify apps. Second, the current app will only link to the apps specified (here app2), if the app has already been specified in a 1 of 2 other ways. <br/>
```

Comment:
When referring to spec keys, we should use backticks to indicate that they're semantically meaningful. `conditional_links`